gitops/generate_build.go

```diff
@@ -14,36 +14,75 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package controllers
+package gitops
 
 import (
 	"encoding/json"
+	"path/filepath"
 	"strings"
 
 	routev1 "github.com/openshift/api/route/v1"
 	appstudiov1alpha1 "github.com/redhat-appstudio/application-service/api/v1alpha1"
+	"github.com/redhat-appstudio/application-service/gitops/resources"
+	yaml "github.com/redhat-appstudio/application-service/gitops/yaml"
+	"github.com/spf13/afero"
 	tektonapi "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
 	triggersapi "github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-func determineBuildDefinition(component appstudiov1alpha1.Component) string {
-	return "devfile-build"
-}
+const (
+	buildCommonStoragePVCFileName = "common-storage-pvc.yaml"
+	buildTriggerTemplateFileName  = "trigger-template.yaml"
+	buildEvetListenerFileName     = "event-listener.yaml"
+	buildWebhookRouteFileName     = "build-webhook-route.yaml"
+)
 
-func determineBuildCatalog(namespace string) string {
-	// TODO: If there's a namespace/workspace specific catalog, we got
-	// to respect that.
-	return "quay.io/redhat-appstudio/build-templates-bundle@sha256:2205a29208fa686b47f841819f7abedb64adb93935493693892d0e18bbdbb77e"
+func GenerateBuild(fs afero.Fs, outputFolder string, component appstudiov1alpha1.Component) error {
+	// Because GenerateAndPush cleans up everything for component it cleans previously generated main gitops files for the component.
```

Comment:
I wonder if we should still do it once though, maybe we can streamline the basic kube and gitops resource generation in one function and call it strategically once in the controller..

Is there a case where we want the basic deployment resource generated but not the gitops resources? If not, then should we pursue it?

Reply:
I was thinking about it. Talked with @sbose78 and he adviced to keep build resources in a separate commit, so if for some reason generation of build gitops resources fails, it won't break main ones.

Reply:
@maysunfaisal @mmorhun I wanted to keep it separate so that the failure of one doesn't impact the other. However, I don't have any other strong reasons for that. I'm open to doing things differently if one of you have strong feelings either way :)

Reply:
A way with two commits is already implemented in this PR

Reply:
But I still have no strong opinion on this

Reply:
Is there a reason why the generation of the build resources would fail that we would want to keep it separate? If it's not likely, I'm inclined to agree with Maysun and would favour doing it all at once.